### PR TITLE
Create a mechanism to modify the beacon version field.

### PR DESCRIPTION
**Pull request by pvanhorn**
This change allows a system to modify the version field in the beacon when the join flag is set.

### CODE REVIEW

File: src/core/mac/mac_frame.hpp [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #ifndef MAC_FRAME_HPP_
 #define MAC_FRAME_HPP_
 
+#include <openthread-core-config.h>
```
Comment (jwhui) [inline src/core/mac/mac_frame.hpp]: Change this to:

```
#ifdef OPENTHREAD_CONFIG_FILE
#include OPENTHREAD_CONFIG_FILE
#else
#include <openthread-config.h>
#endif
```

Example: [src/core/openthread.cpp](https://github.com/openthread/openthread/blob/master/src/core/openthread.cpp)

File: src/core/openthread-core-default-config.h [hunk 2]
```diff
@@ -238,6 +238,16 @@
 #endif  // OPENTHREAD_CONFIG_COAP_MAX_RETRANSMIT
 
 /**
+ * @def OPENTHREAD_CONFIG_JOIN_BEACON_VERSION
+ *
+ * The Beacon version to use when the beacon join flag is set.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_JOIN_BEACON_VERSION
+#define OPENTHREAD_CONFIG_JOIN_BEACON_VERSION kProtocolVersion
```
Comment (jwhui) [inline src/core/openthread-core-default-config.h]: nit: vertical align with values above

File: src/core/mac/mac_frame.hpp (hunk 1 above)
Comment (pvanhorn) [inline src/core/mac/mac_frame.hpp]: I don't see how openthread-config.h includes openthread-core-config.h

Comment (jwhui) [inline src/core/mac/mac_frame.hpp]: Ah, yes, read this too quickly.

File: src/core/openthread-core-default-config.h [hunk 3]
```diff
@@ -238,6 +238,16 @@
...
 
 /**
+ * @def OPENTHREAD_CONFIG_JOIN_BEACON_VERSION
+ *
+ * The Beacon version to use when the beacon join flag is set.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_JOIN_BEACON_VERSION
+#define OPENTHREAD_CONFIG_JOIN_BEACON_VERSION                   kProtocolVersion
+#endif  // OPENTHREAD_CONFIG_ALT_BEACON_VERSION
```
Comment (jwhui) [inline src/core/openthread-core-default-config.h]: nit: `OPENTHREAD_CONFIG_ALT_BEACON_VERSION` -> `OPENTHREAD_CONFIG_JOIN_BEACON_VERSION`

**Pull request merged**